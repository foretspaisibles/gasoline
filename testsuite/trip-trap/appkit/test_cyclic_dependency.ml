(* Test_cyclic_dependency -- Test cyclic module dependency

Author: Michael Grünewald
Date: Sun Sep 21 00:26:03 CEST 2014

Gasoline (https://github.com/michipili/gasoline)
This file is part of Gasoline

Copyright © 2014 Michael Grünewald

This file must be used under the terms of the CeCILL-B.
This source file is licensed as described in the file COPYING, which
you should have received as part of this distribution. The terms
are also available at
http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.txt *)
module Application =
  Gasoline_Plain_Application

let bike =
  Application.Component.make
    ~name:"bike"
    ~require:["cycle"]
    ~description:"A simple bike requiring a cycle"
    ()

let cycle =
  Application.Component.make
    ~name:"cycle"
    ~require:["bike"]
    ~description:"A cycle requiring a bike"
    ()

let () =
  Application.run
    "test_cyclic_dependency" "" ""
    ignore
