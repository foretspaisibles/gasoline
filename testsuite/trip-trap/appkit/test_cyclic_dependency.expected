test_cyclic_dependency.byte: Failure: Cyclic dependency on component 'bike'.
